### PR TITLE
Add data layer for atomic/transfers endpoint

### PR DESCRIPTION
**Pull request by obenland**
#### Changes proposed in this Pull Request

* Addss data layer for atomic/transfers endpoint

#### Testing instructions

* TBD

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/atomic/transfer/index.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+		action
+	);
+
+const showErrorNotice = error => {
+	if ( error.error === 'invalid_input' ) {
+		return errorNotice( translate( 'The uploaded file is not a valid zip.' ) );
+	}
+
+	if ( error.error === 'api_success_false' ) {
+		return errorNotice( translate( 'The uploaded file is not a valid plugin or theme.' ) );
```
Comment (a8ci18n) [inline client/state/data-layer/wpcom/sites/atomic/transfer/index.js]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'The uploaded file is not a valid plugin.' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/state/data-layer/wpcom/sites/atomic/transfer/index.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	}
+
+	if ( error.error ) {
+		return errorNotice(
+			translate( 'Upload problem: %(error)s.', {
+				args: { error: error.error },
+			} )
+		);
+	}
+	return errorNotice( translate( 'Problem uploading the plugin or theme.' ) );
```
Comment (a8ci18n) [inline client/state/data-layer/wpcom/sites/atomic/transfer/index.js]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Problem uploading the plugin.' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/state/data-layer/wpcom/sites/atomic/transfer/index.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	);
+};
+
+/**
+ * Retreives a list of Atomic transfer records.
+ *
+ * @param {?Object} action default action to call on HTTP events
+ * @returns {Object} action object
+ */
+export const getTransfers = action =>
```
Comment (jhnstn) [inline client/state/data-layer/wpcom/sites/atomic/transfer/index.js]: I'm not sure we need to fetch all the transfers for a given site. 
All we should be interested in is the latest transfer which the `sites/:site/transfers/latest` api endpoint should provide for us.
We should however consider moving that to `sites/:site/atomic/transfers/latest`

Comment (obenland) [reply client/state/data-layer/wpcom/sites/atomic/transfer/index.js]: This just corresponds to the default behavior for GET requests to that endpoint. 

> We should however consider moving that to sites/:site/atomic/transfers/latest

Good news, it's already there! 🎉
https://opengrok.a8c.com/source/xref/trunk/wp-content/rest-api-plugins/endpoints/sites-atomic-transfers.php#245

I'll add the corresponding data layer

File: client/state/data-layer/wpcom/sites/atomic/transfers/index.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+		action
+	);
+
+const showErrorNotice = error => {
+	if ( error.error === 'invalid_input' ) {
+		return errorNotice( translate( 'The uploaded file is not a valid zip.' ) );
+	}
+
+	if ( error.error === 'api_success_false' ) {
+		return errorNotice( translate( 'The uploaded file is not a valid plugin or theme.' ) );
```
Comment (a8ci18n) [inline client/state/data-layer/wpcom/sites/atomic/transfers/index.js]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'The uploaded file is not a valid plugin.' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/state/data-layer/wpcom/sites/atomic/transfers/index.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+	}
+
+	if ( error.error ) {
+		return errorNotice(
+			translate( 'Upload problem: %(error)s.', {
+				args: { error: error.error },
+			} )
+		);
+	}
+	return errorNotice( translate( 'Problem uploading the plugin or theme.' ) );
```
Comment (a8ci18n) [inline client/state/data-layer/wpcom/sites/atomic/transfers/index.js]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Problem uploading the plugin.' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by obenland** [comment]
@jhnstn This should be working now.
To test without actually transferring the site, comment out the Transfer_Manager bits in the endpoint's `create_item` method and return `$args`.

**Comment by obenland** [comment]
@dmsnell Does this look sane? First time I added something to `data-layer`.

**Comment by jhnstn** [comment]
Verified the ecommerce flow is working.

**Comment by dmsnell** [comment]
@obenland I'm coming back from vacation. are there things you'd like for me to review specifically at this point? are you still working on this? should I do a new review fresh from the get-go?

**Comment by obenland** [comment]
@dmsnell Thanks for checking in! A month ago it seemed like it needed to get in ASAP, now I'm not sure when it will. Let me ping you again when it becomes clear when that will be, and what exactly that data layer then needs to look like!

**Pull request closed without merging**